### ISSUE TITLE
Meshcentral suddenly stopped to automatically renew the Let's Encrypt certificate

### ISSUE THREAD

**Issue by LG00999** [Ylianst/MeshCentral#4676]
Dear all, 

We have a virtual machine on which Meshcentral is installed and works flawessly for months. 
Suddendly, few weeks ago, it seems the certificate automatic renewal procedure has stop working.
Into the Meshcentral console, we got this error message:
"**Failed to obtain certificate: Error finalizing order :: signature algorithm not supported**"

According to the research I've made on internet, it seems there an issue with some algorithm during the communication between our MeshCentral server and Let's Encrypt.
Does anyone has a clue on how to fix this problem?

Technical informations:
The VM is a Windows 10 computer
the Meshcentral version is 0.6.24 (we know that we should upgrade but since we have more than 600 computers on it and that it was working well, we were a bit afraid to upgrade).

Any help is welcomed.
Thank you very much.

LG

**Comment by si458** [comment]
You are running a very old version which uses an old cryptography that let's Encrypt no longer supports

You need to upgrade to the latest version which supports the new cryptography

**Comment by dinger1986** [comment]
good catch @si458 @LG00999 you can also check out the SSL docs here https://ylianst.github.io/MeshCentral/meshcentral/SSLnletsencrypt/

**Comment by LG00999** [comment]
Thank you for your answer.
Like I said, we know that we have to upgrade but prior to do this we need to make a lab , some préproduction test, etc...so as to be sure that everything will work as expected with this upgrade. 
The problem is that our Let's encrypt certificate will expired very soon (in 6 days).
So my question is: by waiting this upgrade, is there a workaround (some configuration modification?) which would allow our MeshCentral server to communicate again with Let's encrypt server with the good algorithms/protocols ?
Thank you.

LG

**Comment by Ylianst** [comment]
From time to time to handle new security threats Let's encrypt will update it's API and remove support for the old ones. Because your running such an old version of MeshCentral, it's likely that you ran into this.

Without updating I see two solutions. You can buy a commercial TLS certificate from GoDaddy or another vendor and install it in MeshCentral. You will need to replace the following two files in the `meshcentral-data` folder with your new certificate:

```
webserver-cert-public.crt
webserver-cert-private.key
```

There two files are text files, so use a text editor to look at them and replace them with sililar looking files. You then disable Let's Encrypt support in MeshCentral, you just add "_" in front of the "letsencrypt" entry in the config.json. That will do it.

Another option is to use CertBot, a Let's Encrypt tools to get a certificate and then do the same thing as above to setup the certificate manually in MeshCentral. In that case, there are instructions online on how to use CertBot.

Obviously, always keep a full offsite backup of your server before doing anything. The backup must include the database and "meshcentral-data" folder at a minimum.

Hope this helps.

**Comment by LG00999** [comment]
Hello Ylian, 

Thank you very much for your help. I think buying a commercial certificate will be the best solution for now and will buy us some time to prepare the upgrade.
Regarding this upgrade, I have some questions about it. I'm not sure if I have to open a new thread for it.
My main questions are:
- Is there known issue(s) when upgrading from 0.6.24 to 1.0.0?
- what's happening to the client side (agent) when upgrading the server? Are thoses agents will upgrade themselves automatically? Could there be issue for this part? (e.g. agents won't upgrade for some reason). This is crucial for us because we have more than 600 agents deployed to this day.
- I intend to make a Vmware snapshot of our Meshcentral server prior to the upgrade. In case something is going wrong during this upgrade (server side or agent sides), will it be possible to get back to the previous state of the server and can we hope that everything will work the way it was before the upgrade?
-Thanks again for your precious help (and by the way, you're doing a incredible job with Meshcentral)

LG

**Comment by Ylianst** [comment]
On the questions.

 - 0.6.24 to 1.0.0 is a HUGE jump. It should work I have never tested such a big jump. If you have a large production system or this is critical infrastructure, I recommend you first test it on a pre-production server. The other option is to make smaller jumps before going to 1.0.0.
 - Yes, once the server is updated it will update all of the agents. This is the most critical part and where things could go wrong and be a disaster. You can always roll back the server, but if the agent updates fail, there is no way to remotely fix it. One idea would be to block all incoming connections except a small IP address range with machines you have access to, update the server and make sure they all update right.
 - Good. It's always important to keep offsite updates, especially the "meshcentral-data" folder and your database if your using MongoDB or something. Once you have a backup, in general you just need to copy "node_modules" into a backup folder. If things go wrong, just put the old "node_modules" back and the old server will be back.

Another things you can do is this:

```
{
  "settings": {
    "noAgentUpdate": 1
  }
}
```

This will tell the server to not update the agents, but this may cause issues of it's own since the old agent will get told to try to run the new meshcore and it may not work, but you can always roll back the server. I am not going to go into the details of this, but pre-production testing would be best.

**Comment by ctrlaltca** [comment]
Hi, I'm getting the same error on meshcentral 1.0.0:
```
meshcentral@<hidden>:~$ node node_modules/meshcentral --debug cert
MeshCentral HTTP redirection server running on port 80.
CERT: LE: Getting certs from local store (Production)
CERT: LE: Reading certificate files
CERT: LE: Setting LE cert for default domain.
MeshCentral v1.0.0, WAN mode.
MeshCentral Intel(R) AMT server running on <hidden>:4433.
MeshCentral HTTPS server running on <hidden>:443.
CERT: LE: Certificate has 18 day(s) left.
CERT: LE: Asking for new certificate because of expire time.
CERT: LE: Generating private key...
CERT: LE: Setting up ACME client...
CERT: LE: Creating certificate request...
CERT: LE: Requesting certificate from Let's Encrypt...
CERT: LE: Succesful response to challenge.
CERT: LE: Succesful response to challenge.
CERT: LE: Succesful response to challenge.
CERT: LE: Succesful response to challenge.
CERT: LE: Succesful response to challenge.
CERT: LE: Failed to obtain certificate: Error finalizing order :: signature algorithm not supported
```
Last successful renew of the certificate was on august 20.
From a quick search it could be releted to this https://community.letsencrypt.org/t/unable-to-validate-my-domain-since-the-last-couple-of-days-due-to-signature-algorithm-not-supported-it-was-working-fine-before/184731 and this: https://github.com/digitalbazaar/forge/issues/986

**Comment by LG00999** [comment]
Thank you very much Ylian for your detailed answer.
I have one last question: what would be your advice so as to update from 0.6.24 to 1.0? What intermediate versions you think we have to go through before applying the 1.0? (and where to find those interemediate versions?).
Many thanks again.

LG

**Comment by trentdavis** [comment]
I just started having the same issue, but I'm using version 1.0.0.  My problem is, this is a production server, so I try to stick to the latest stable version.  However, Letsencrypt has served me well, so I'd like to keep using it.  Is it ok to upgrade to the latest version?

**Comment by bmercernccer** [comment]
@trentdavis I had the same issue. I've applied 1.1.0 to resolve it. so far it's working ok, nothing has blown up yet.

**Comment by TakoyakiAllenC** [comment]
I also updated to 1.0.0 and it failed to resolve the issue. 1.1.0 seems to have resolved it!

**Comment by PR-JF** [comment]
Had the same issue -- confirmed 1.1.0 resolved. We also are running in production environment, and so ideally would like to stick with "stable" version. Having a LTS version with clear release schedule would be ideal.

**Comment by dinger1986** [comment]
@si458 this is fixed

**Comment by si458** [comment]
fixed in 1.1.0 or above
